piper/ratbagd.py

```diff
@@ -186,6 +186,27 @@ def _set_dbus_property(self, property, type, value, readwrite=True):
         # update
         self._proxy.set_cached_property(property, val)
 
+    def _dbus_call_async(self, method, type, *value, callback):
+        # The callback handler to unwrap the data
+        def callback_handler(obj, result, user_callback):
+            res = obj.call_finish(result)
+            res = res.unpack()[0]  # Result is always a tuple
+            user_callback(res)
+
+        # Calls a method asynchronously on the bus, using the given method
+        # name, type signature and values. The callback is invoked when the
+        # method finishes
+        val = GLib.Variant("({})".format(type), value)
+        try:
+            self._proxy.call(method, val,
+                             Gio.DBusCallFlags.NO_AUTO_START,
+                             500, None, callback_handler, callback)
+        except GLib.Error as e:
+            # Unrecognized error code; print the message to stderr and raise
+            # the GLib.Error.
+            print(e.message, file=sys.stderr)
+            raise
```

Comment:
Do we have an idea what errors can be raised here? It'd be better to try and catch them. The documentation lists the following but IIRC there are others too:

> If the `Gio.DBusConnection` associated with `self` is closed then the operation will fail with `Gio.IOErrorEnum.CLOSED`. If `cancellable` is canceled, the operation will fail with `Gio.IOErrorEnum.CANCELLED`. If parameters contains a value not compatible with the D-Bus protocol, the operation fails with `Gio.IOErrorEnum.INVALID_ARGUMENT`.